go/chat/server.go

```diff
@@ -671,6 +670,9 @@ func (h *Server) GetThreadNonblock(ctx context.Context, arg chat1.GetThreadNonbl
 
 	// Apply any pager mode transformations
 	pagination = h.applyPagerModeIncoming(ctx, arg.ConversationID, pagination, arg.Pgmode)
+	if pagination != nil && pagination.Last {
```

Comment:
What is this for?

Reply:
if we get a null result from `Pull` our outgoing pagination is set to `&chat1.Pagination{Num:x, Prev:0, Next:0, Last:true}`, if we call the function again Pull will default to give use the `x` results since `Next` is unset so i wanted to short circuit here